drivers/i2c/i2c_mspm0g3xxx.c

```diff
@@ -249,6 +250,27 @@ static int i2c_mspm0g3xxx_configure(const struct device *dev, uint32_t dev_confi
 	case I2C_SPEED_FAST:
 		bitrate = 7;
 		break;
+	case I2C_SPEED_DT:
+		// First figure out if the number divides perfectly
+		if (config->parent_clock_freq % (10 * config->clock_frequency)) {
```

Comment:
Also you could move some of the checking here into the preprocessor. Any error in the device tree with fail to compile rather than shipping a broken image